mmdet/datasets/pipelines/transforms.py

```diff
@@ -1809,3 +1809,271 @@ def __repr__(self):
+    (0:cut_y, cut_x:self.size[1])             -- image 1
+    (cut_y:self.size[0], 0:cut_x)             -- image 2
+    (cut_y:self.size[0], cut_x:self.size[1])  -- image 3
+    The output image is composed of the parts from each sub-image.
+    Args:
+        size (tuple): Output size of the image. Default: (640, 640).
+        jitter (float): It decides the size of the cropping window.
+        min_offset (float): Volume of the offset of the cropping window.
+        load_pipeline (list): Pipeline for loading images and annotations.
+        letter_box(bool): Whether to maintain the aspect ratio of the
+            subimages.
+    """
+
+    def __init__(self,
+                 size=(640, 640),
+                 min_offset=0.2,
+                 dataset=None,
+                 allow_negative_crop=False,
+                 bbox_clip_border=True):
+        assert isinstance(size, tuple)
+        assert size[0] > 0 and size[1] > 0, (
+            'size must > 0 in train mode')
+        assert isinstance(size[0], int) and isinstance(
+            size[1], int)
+
+        if min_offset is None:
+            self.min_offset = (0, 0)
+        else:
+            if isinstance(min_offset, float):
+                assert 0 <= min_offset <= 1
+                self.min_offset = (min_offset, min_offset)
+            elif isinstance(min_offset, tuple):
+                assert isinstance(min_offset[0], float) \
+                       and isinstance(min_offset[1], float)
+                assert 0 <= min_offset[0] <= 1 and \
+                       0 <= min_offset[1] <= 1
+                self.min_offset = min_offset
+
+        self.size = size
+
+        self.allow_negative_crop = allow_negative_crop
+        self.dataset = dataset
+        self.bbox_clip_border = bbox_clip_border
+        self.bbox2label = {
+            'gt_bboxes': 'gt_labels',
+            'gt_bboxes_ignore': 'gt_labels_ignore'
+        }
+
+    def __call__(self, results):
+        """Call the function to mix 4 images
+        Args:
+            results (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: Result dict with images and bounding boxes cropped.
+        """
+        # Generate the Mosaic coordinate
+        cut_y = random.randint(
+            int(self.size[0] * self.min_offset[0]),
+            int(self.size[0] * (1 - self.min_offset[0]))
+        )
+        cut_x = random.randint(
+            int(self.size[1] * self.min_offset[1]),
+            int(self.size[1] * (1 - self.min_offset[1]))
+        )
+
+        buffer_img = np.zeros((self.size[0], self.size[1], 3), dtype=np.float32)
+
+        out_bboxes = []
+        out_labels = []
+        out_ignores = []
+
+        for i in range(4):
+            if i == 0:
+                # use the current image
+                results_i = results
+            else:
+                # randomly sample a new image from the dataset
+                results_i = next(self.dataset)
```

Comment:
Better use `np.random.randint(0, len(self.dataset), size=3)` before the `for` loop.

Reply:
fixed